stdlib/public/core/KeyPath.swift

```diff
@@ -653,6 +658,8 @@ internal enum KeyPathComponent: Hashable {
   case optionalForce
   /// The keypath wraps a value in an optional.
   case optionalWrap
+  /// The keypath references an enum case with an associated value.
+  case payloadCase(tag: Int)
```

Comment:
@slavapestov Is an integer tag sufficient to address a non-frozen enum case externally? We'd want to test that references to a public case from inside and outside the module have matching `==` identity as well.

Reply:
Cases can be re-ordered resiliently, so as long as the tag is obtained by loading from the enum case tag global and not hard-coded in the client, then it should be fine.

Reply:
You should add some tests to validation-test/Evolution/ to check ABI compatibility with the same client and a library built twice with changes. I see there is already a keypath test there.

Reply:
OK, so it sounds like the instantiation pattern should use a string, but we can instantiate that into the case index in the object that gets instantiated at runtime.

Reply:
A string might work, but I was thinking an indirect reference to an enum case tag global (which is emitted in the same module as the enum itself).